src/Features/LanguageServer/Protocol/Handler/Completion/CompletionHandler.cs

```diff
@@ -114,59 +148,131 @@ bool IsValidTriggerCharacterForDocument(Document document, char triggerCharacter
+                    var roslynCompletionItem = await CreateCompletionItemAsync<LSP.CompletionItem>(
+                        request, document, item, resultId, completionTrigger, commitCharacterRulesCache,
+                        completionService, clientName, returnTextEdits, documentText, defaultSpan,
+                        defaultRange, cancellationToken).ConfigureAwait(false);
                     return roslynCompletionItem;
                 }
             }
 
-            static TCompletionItem CreateCompletionItem<TCompletionItem>(
+            static async Task<TCompletionItem> CreateCompletionItemAsync<TCompletionItem>(
                 LSP.CompletionParams request,
+                Document document,
                 CompletionItem item,
                 long? resultId,
                 CompletionTrigger completionTrigger,
-                Dictionary<ImmutableArray<CharacterSetModificationRule>, ImmutableArray<string>> commitCharacterRulesCache) where TCompletionItem : LSP.CompletionItem, new()
+                Dictionary<ImmutableArray<CharacterSetModificationRule>, ImmutableArray<string>> commitCharacterRulesCache,
+                CompletionService completionService,
+                string? clientName,
+                bool returnTextEdits,
+                SourceText? documentText,
+                TextSpan? defaultSpan,
+                LSP.Range? defaultRange,
+                CancellationToken cancellationToken) where TCompletionItem : LSP.CompletionItem, new()
             {
                 var completeDisplayText = item.DisplayTextPrefix + item.DisplayText + item.DisplayTextSuffix;
+
+                // The TextEdits for override and partial method completions are provided in the resolve handler.
+                // We do not provide them in this handler.
+                item.Properties.TryGetValue("Modifiers", out var modifier);
+                var isOverrideOrPartialMethodCompletion = modifier != null && (modifier.Contains("Override") || modifier.Contains("Partial"));
+
+                LSP.TextEdit? textEdit = null;
+                if (returnTextEdits && !isOverrideOrPartialMethodCompletion)
+                {
+                    textEdit = await GenerateTextEdit(
+                        document, item, completionService, documentText, defaultSpan, defaultRange, cancellationToken).ConfigureAwait(false);
+                }
+
                 var completionItem = new TCompletionItem
                 {
+                    TextEdit = returnTextEdits
+                        ? textEdit
+                        : null,
+                    InsertText = returnTextEdits || isOverrideOrPartialMethodCompletion
+                        ? null
+                        : item.Properties.ContainsKey("InsertionText") ? item.Properties["InsertionText"] : completeDisplayText,
+                    InsertTextFormat = isOverrideOrPartialMethodCompletion && clientName == "RazorCSharp"
```

Comment:
Snippet support is only valid for Razor at the moment. We could enable it for C# once LSP completion is enabled in local.

Reply:
we really should not just hardcode in strinsg like this.  If this name changes elsewhere, this just breaks.  Can we reference an appropriate constant defined in a single location?

Reply:
If you want to magically enable it for C# you could; however, it wouldn't actually work in LiveShare scenarios, just local LSP:

Per our offline conversations I added C# to the list of supported experimental users
https://devdiv.visualstudio.com/DevDiv/_git/VSLanguageServerClient/pullrequest/305629?_a=files&path=%2Fsrc%2Fproduct%2FRemoteLanguage%2FImpl%2FFeatures%2FCompletion%2FExperimentalSnippetSupport.cs&discussionId=3138420